examples/hybrid_jobs/6_QNSPSA_optimizer_with_embedded_simulator/qnspsa_with_embedded_simulator.ipynb

```diff
@@ -0,0 +1,1360 @@
+{
```

Comment:
<p>I might be wrong, but I thought parameter-shift depends on num of parameters not the num of gates</p><p> </p><p>Also,</p><blockquote>On the other hand, stochastic methods always execute a fixed number of circuits for each step.</blockquote><p>Not suggesting adding to the notebook. I am just interested what is the "fixed number"? And how does the number of gates scale with number of qubits for gradient-based&nbsp;methods</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-braket-examples/pull/149/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

Reply:
<p>Parameter-shift depends on the number of parametrized gates.</p>

Reply:
@yitchen-tim Step-wise, SPSA takes 2 circuits for gradient estimation. QNSPSA does the same gradient estimation, and 4 more circuits for the tensor metric. Yet the 4 circuits for the tensor metrics are twice as deep as the other ones. The blocking setting in QNSPSA requires an additional two circuits measuring the current and the next step loss value.  To sum up with a hand-wavy argument: a single step of QNSPSA should take 2 + 4*2 + 2 = 12X the execution time of a qnode for the problem.